toxcore/tox_api.c

```diff
@@ -164,9 +165,34 @@ const char *tox_options_get_proxy_host(const Tox_Options *options)
 {
     return options->proxy_host;
 }
-void tox_options_set_proxy_host(Tox_Options *options, const char *proxy_host)
+bool tox_options_set_proxy_host(Tox_Options *options, const char *proxy_host)
 {
-    options->proxy_host = proxy_host;
+    if (!options->experimental_owned_data) {
+        options->proxy_host = proxy_host;
+        return true;
+    }
+
+    if (options->owned_proxy_host != nullptr) {
+        free(options->owned_proxy_host);
+        options->owned_proxy_host = nullptr;
+    }
+    if (proxy_host == nullptr) {
+        options->proxy_host = nullptr;
+        return false;
+    }
+
+    const size_t proxy_host_length = strlen(proxy_host) + 1;
+    char *owned_ptr = (char *)malloc(proxy_host_length);
+    if (owned_ptr != nullptr) {
+        memcpy(owned_ptr, proxy_host, proxy_host_length);
+        options->proxy_host = owned_ptr;
+        options->owned_proxy_host = owned_ptr;
+    } else {
+        options->proxy_host = proxy_host;
+        options->owned_proxy_host = nullptr;
```

Comment:
Should this branch return `false` to indicate that it doesn't own the data?

Reply:
Yes, fixed. Thanks :).